/**/controller_manager:
  ros__parameters:
    update_rate: 100 # Hz

    joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    gripper_controller:
      type: position_controllers/GripperActionController

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

/**/joint_trajectory_controller:
  ros__parameters:
    joints:
      - $(var tf_prefix)joint_1
      - $(var tf_prefix)joint_2
      - $(var tf_prefix)joint_3
      - $(var tf_prefix)joint_4
      - $(var tf_prefix)joint_5
      - $(var tf_prefix)joint_6
    command_interfaces:
      - position
      - velocity
    state_interfaces:
      - position
      - velocity

    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    allow_nonzero_velocity_at_trajectory_end: true
    constraints:
      stopped_velocity_tolerance: 0.02
      goal_time: 0.0
      $(var tf_prefix)joint_1: { trajectory: 0.2, goal: 0.02 }
      $(var tf_prefix)joint_2: { trajectory: 0.2, goal: 0.02 }
      $(var tf_prefix)joint_3: { trajectory: 0.2, goal: 0.02 }
      $(var tf_prefix)joint_4: { trajectory: 0.2, goal: 0.02 }
      $(var tf_prefix)joint_5: { trajectory: 0.2, goal: 0.02 }
      $(var tf_prefix)joint_6: { trajectory: 0.2, goal: 0.02 }

    gains: # Required because we're controlling a velocity interface
      $(var tf_prefix)joint_1: { p: 10.0, d: 1.0, i: 1.0, i_clamp: 1.0 } # Smaller 'p' term, since ff term does most of the work
      $(var tf_prefix)joint_2: { p: 10.0, d: 1.0, i: 1.0, i_clamp: 1.0 }
      $(var tf_prefix)joint_3: { p: 10.0, d: 1.0, i: 1.0, i_clamp: 1.0 }
      $(var tf_prefix)joint_4: { p: 10.0, d: 1.0, i: 1.0, i_clamp: 1.0 }
      $(var tf_prefix)joint_5: { p: 10.0, d: 1.0, i: 1.0, i_clamp: 1.0 }
      $(var tf_prefix)joint_6: { p: 10.0, d: 1.0, i: 1.0, i_clamp: 1.0 }

    velocity_ff:
      $(var tf_prefix)joint_1: 1.0
      $(var tf_prefix)joint_2: 1.0
      $(var tf_prefix)joint_3: 1.0
      $(var tf_prefix)joint_4: 1.0
      $(var tf_prefix)joint_5: 1.0
      $(var tf_prefix)joint_6: 1.0

/**/gripper_controller:
  ros__parameters:
    joint: $(var tf_prefix)gripper_jaw1_joint
